include/boost/math/differentiation/autodiff.hpp

```diff
@@ -9,8 +9,14 @@
 #include <boost/config.hpp>
 #include <boost/math/tools/config.hpp>
 
+#if defined(_MSC_VER) || defined(BOOST_MSVC)
+#define NOMINMAX
```

Comment:
#define in a header file is not ideal. Is this to overcome the ridiculous #defines of min and max in the windows headers? If so could the same be achieved by enclosing our calls to min/max in parentheses? i.e. ```auto m = (std::max)(a, b)```. I think I saw that pattern used elsewhere in this file (or at least this repo).

Reply:
These files are in beta stage at best. Before time of submission, necessary but messy code will move, and the rest will be removed.

That #define is related to Windows max/min, for which I’m using parens where needed.

Reply:
I removed the code you highlighted.